ENH: Add overlapping volume mosaic screenshot script

jhlegarreta opened the pull request:
- ENH: Add overlapping volume mosaic screenshot script
- ENH: Remove unused imports
- ENH: Remove unused local variable
- ENH: Allow using a transparency mask in fODF visualization
- ENH: Use `assert_same_resolution` in fODF visualization script checks

pep8speaks commented:
Hello @jhlegarreta, Thank you for submitting the Pull Request !







* In the file [`scripts/scil_screenshot_volume_mosaic_overlap.py`](https://github.com/scilus/scilpy/blob/7b90d7e96ae208a6598a988bde9a57aedbc114d0/scripts/scil_screenshot_volume_mosaic_overlap.py):

> [Line 167:80](https://github.com/scilus/scilpy/blob/7b90d7e96ae208a6598a988bde9a57aedbc114d0/scripts/scil_screenshot_volume_mosaic_overlap.py#L167): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (81 > 79 characters)
> [Line 176:80](https://github.com/scilus/scilpy/blob/7b90d7e96ae208a6598a988bde9a57aedbc114d0/scripts/scil_screenshot_volume_mosaic_overlap.py#L176): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (89 > 79 characters)

* In the file [`scripts/scil_visualize_fodf.py`](https://github.com/scilus/scilpy/blob/7b90d7e96ae208a6598a988bde9a57aedbc114d0/scripts/scil_visualize_fodf.py):

> [Line 194:80](https://github.com/scilus/scilpy/blob/7b90d7e96ae208a6598a988bde9a57aedbc114d0/scripts/scil_visualize_fodf.py#L194): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (100 > 79 characters)

scil-admin commented:
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-577/2/

scil-admin commented:
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-577/3/

scil-admin commented:
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-577/4/

frheault commented:
Can we have an visual example (screenshot) of a command and the results (for  Add overlapping volume mosaic screenshot script)?

jhlegarreta commented:
> Can we have an visual example (screenshot) of a command and the results (for Add overlapping volume mosaic screenshot script)?

The screenshots added should be enough I'd say. The examples in the script header documentation were used to generate them, and they are included in the dedicate documentation file.

karanphil commented:
Nice additions! I tested both the transparency option on `scil_visualize_fodf.py` and the mosaic script. They work very well. Thanks!

frheault commented:
Ah I see, the image did not load on my phone, they are perfect! Sorry about that.

arnaudbore commented on scripts/scil_visualize_fodf.py:
```diff
@@ -171,6 +175,13 @@ def _get_data_from_inputs(args):
     between the data for mask, background, peaks and fODF.
     """
+
+    def _check_mask_shape(_mask, _fodf):
```
Use assert_same_resolution don't need extra function

jhlegarreta replied:
Done in f87598e. Note that I re-used what was being done for other arguments.

arnaudbore commented on scilpy/viz/scene_utils.py:
```diff
@@ -481,4 +486,406 @@ def render_scene(scene, window_size, interactor,
+    return image
+
+
+def draw_scene_at_pos(
```
You can use the full line for the args of the def.

jhlegarreta replied:
Automatic code formatting breaks it this way. Avoids the user needing to manually break, the user breaking the line at times when the full length has not been used, etc.

arnaudbore reviewed (commented):
A couple of comments, I need to test it. I don't know if it's a consensus but I feel more confortable to have axial screenshots with the frontal lob oriented to the top.

jhlegarreta commented on scripts/scil_screenshot_volume_mosaic_overlap.py:
```diff
@@ -0,0 +1,240 @@
+
+    add_overwrite_arg(p)
+
+    return p
```
The scripts are not consistent at this.

jhlegarreta commented:
> I don't know if it's a consensus but I feel more confortable to have axial screenshots with the frontal lob oriented to the top.

If the flip is not applied, the sagittal and coronal slices are upside-down.

CHrlS98 commented on scripts/scil_screenshot_volume_mosaic_overlap.py:
```diff
@@ -0,0 +1,240 @@
+    if args.in_labelmap:
+        inputs.append(args.in_labelmap)
+
+    output.append(args.out_fname)
```
I'm not against having this method (of course since I wrote both examples Jon cites). Can be particularly useful when the number of optional arguments grows bigger and there starts being dependencies between some of them. Same goes for `_get_data_from_inputs`. If you look in [`sci_visualize_fodf.py`](https://github.com/scilus/scilpy/blob/933ae303ca51ba2f95e1da3cf6a93a5a4f6f6c83/scripts/scil_visualize_fodf.py#L169) it turned into a pretty big method that does simplify the `main` in my opinion.

But it is true that for the actual script it does not simplify the `main` that much (at least for now, I don't know if there are plans to add other options in the future).

scil-admin commented:
Build passed ! Good Job :beers: !

jhlegarreta commented on scripts/scil_screenshot_volume_mosaic_overlap.py:
```diff
@@ -0,0 +1,240 @@
+    parser = _build_arg_parser()
+    args = _parse_args(parser)
+
+    vol_img, mask_img, labelmap_img = _get_data_from_inputs(args)
```
> Check if header are compatible once you have the whole list files you can use assert_same_resolution

Done in 8db9b0c.

scil-admin commented:
Build passed ! Good Job :beers: !

arnaudbore commented:
Hi @jhlegarreta ,
Sorry for the delay, quick things regarding the formating of your code. We are doing our best to homogenise the way we code in scilpy. I'm aware that the choices we made may be not the best but please follow my instructions. If you don't, I will merge your PR but I'll do the modifications myself right after in another PR...not very efficient.
Thank you in advance

arnaudbore commented:
@jhlegarreta scil_visualize_fodf.py crashes if there is no transparency_mask. I need to run more tests before merging.

arnaudbore commented:
@jhlegarreta please fix scil_visualize_fodf.py.

jhlegarreta commented:
@arnaudbore Done. My mistake. Sorry.

The `test_silent_without_output` was doing its job; I did not check carefully the purpose of it; I've added a test that checks the output when no transparency mask is given.

scil-admin commented:
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-577/8/

jhlegarreta commented:
> I've added a test that checks the output when no transparency mask is given.

Removed, as due to the issues with FURY we are unable to test taking screenshots.

scil-admin commented:
Build passed ! Good Job :beers: !

arnaudbore reviewed (approved):
LGTM
